src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    auto& desc = m_memptr->getDesc();
+    m_element_type = InferenceEngine::details::convertPrecision(desc.getPrecision());
+    m_shape = desc.getShape().toPartialShape().to_shape();
+    get_strides();
+}
+
+void Tensor::set_shape(ov::Shape new_shape) { // this is not as simple as it looks
+// if the ov::Shape is defined, then call cloneWithNewDims to the Memory object memory desc
+// if ov::Shape is undefined (dynamic), we probably have to reintialize the internal Memory object
+// using plain memory format
+    auto& desc = m_memptr->getDesc();
+    const auto newdesc = desc.cloneWithNewDims(new_shape, true);
+    m_memptr->redefineDesc(newdesc);
+
+    m_shape = std::move(new_shape);
+    m_strides.clear();
+    update_strides();
+}
+
+const ov::element::Type& Tensor::get_element_type() const { // trivial -> simply convert the precision type
+    return m_element_type;
+}
+
+const ov::Shape& Tensor::get_shape() const { //trivial -> convert cpu Shape to ov::Shape
+    return m_shape;
+}
+
+size_t Tensor::get_size() const { // trivial -> call to getElementsCount to Shape
+    auto& desc = m_memptr->getDesc();
+    return desc.getShape().getElementsCount();
+}
+
+size_t Tensor::get_byte_size() const { // trivial-> call getCurrentMemSize to memory desc
+    auto& desc = m_memptr->getDesc();
+    return desc.getCurrentMemSize();
+}
+
+const ov::Strides& Tensor::get_strides() const { // here I would ask you to investiage what solution may be better
+// in the case of BlockedMemory desc, we can retrive Strides, but  I'm not sure that the format is the same
+// also it is not clear what to do if the strides does not make sense (some tricky things like Sparced memory)
+// perhaps we are not allowed to wrap any cpu Memory except built over BlockedMemoryDesc
+    std::call_once(m_strides_once, &Tensor::update_strides, this);
+    return m_strides;
+}
+
+void Tensor::update_strides() const {
+    // only support plain data format so far
+    auto blocked_desc = m_memptr->GetDescWithType<BlockedMemoryDesc>();
+    OPENVINO_ASSERT(blocked_desc);
+    OPENVINO_ASSERT(blocked_desc->hasLayoutType(LayoutType::ncsp));
```

Comment:
It does make sense to move the both checks to constructor, also please add error messages.

Reply:
done